components/rhs_card/rhs_card.tsx

```diff
@@ -1,59 +1,71 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import Scrollbars from 'react-custom-scrollbars';
 import {FormattedMessage} from 'react-intl';
 import {Link} from 'react-router-dom';
+import {Post} from 'mattermost-redux/types/posts';
+import {Channel} from 'mattermost-redux/types/channels';
 
 import DelayedAction from 'utils/delayed_action';
-import Constants, {RHSStates} from 'utils/constants';
+import Constants, {RHSStates, ValueOf} from 'utils/constants';
 import * as Utils from 'utils/utils.jsx';
 import RhsCardHeader from 'components/rhs_card_header';
 import Markdown from 'components/markdown';
 import UserProfile from 'components/user_profile';
 import PostProfilePicture from 'components/post_profile_picture';
 import * as GlobalActions from 'actions/global_actions.jsx';
+import {PluginsState} from '../../types/store/plugins';
+
+interface Props {
+    selected?: Post;
+    pluginPostCardTypes: PluginsState['postCardTypes'];
+    previousRhsState?: ValueOf<typeof RHSStates>;
+    enablePostUsernameOverride?: boolean;
+    teamUrl: string;
+    channel?: Channel;
+}
+
+interface State {
+    isScrolling: boolean;
+    topRhsPostCreateAt: number;
+    selected?: Post;
+}
 
-export function renderView(props) {
+export function renderView(props: any) {
     return (
         <div
             {...props}
             className='scrollbar--view'
         />);
 }
 
-export function renderThumbHorizontal(props) {
+export function renderThumbHorizontal(props: any) {
     return (
         <div
             {...props}
             className='scrollbar--horizontal'
         />);
 }
 
-export function renderThumbVertical(props) {
+export function renderThumbVertical(props: any) {
     return (
         <div
             {...props}
             className='scrollbar--vertical'
         />);
 }
 
-export default class RhsCard extends React.Component {
-    static propTypes = {
-        selected: PropTypes.object,
-        pluginPostCardTypes: PropTypes.object,
-        previousRhsState: PropTypes.oneOf(Object.values(RHSStates)),
-        enablePostUsernameOverride: PropTypes.bool,
-        teamUrl: PropTypes.string,
-    }
+export default class RhsCard extends React.Component<Props, State> {
+    state: State;
```

Comment:
`state` is already a property of `React.Component`, you don't need to declare it.

Reply:
I added it based on [this suggestion](https://github.com/typescript-cheatsheets/react-typescript-cheatsheet#class-components) from [react-typescript-cheatsheet](https://github.com/typescript-cheatsheets/react-typescript-cheatsheet) (there is a subsection called 'Why annotate `state` twice?').

I can remove it, if you prefer

Reply:
We usually do it in the constructor if it is required. If we aren't using `state` at all then we just don't declare the type. I haven't run into any issues using `this.setState` with TypeScript so I think we're okay to remove it.